### PR TITLE
Change cache path default

### PR DESCRIPTION
**Pull request by michelleangela**
Change the default cache path from workspace to user folder.

See details in #3664

### CODE REVIEW

File: Extension/src/common.ts [hunk 1]
```diff
@@ -292,6 +292,17 @@ export function isOptionalArrayOfString(input: any): input is string[]|undefined
     return input === undefined || isArrayOfString(input);
 }
 
+export function resolveCachePath(input: string, additionalEnvironment: {[key: string]: string | string[]}): string {
+    let resolvedPath: string = (process.platform === 'win32') ? process.env.USERPROFILE : process.env.HOME;
+    if (!input || input === "") {
```
Comment (WardenGnaw) [inline Extension/src/common.ts]: "" (Empty string) should already be a [falsy](https://basarat.gitbooks.io/typescript/docs/javascript/truthy.html). Where are you doing the check to see if specified path is invalid?

E.g. just
`if (!input)`

File: Extension/package.json [hunk 2]
```diff
@@ -293,8 +293,8 @@
         },
         "C_Cpp.intelliSenseCachePath": {
           "type": "string",
-          "default": "${workspaceFolder}/.vscode",
-          "description": "Defines the folder path for cached precompiled headers used by IntelliSense. The default path \"${workspaceFolder}/.vscode\" will be used if a specified path is invalid.",
+          "default": null,
+          "description": "Defines the folder path for cached precompiled headers used by IntelliSense. The default cache path is \"%USERPROFILE%\/.cpptools\" on Windows and \"~/.cpptools\" on Linux and Mac. The default path will be used if no path is specified or if a specified path is invalid.",
```
Comment (sean-mcmanus) [inline Extension/package.json]: I think we wanted to change this to something more specific like ms-vscode.cpptools.

File: Extension/src/common.ts (hunk 1 above)
Comment (michelleangela) [reply Extension/src/common.ts]: The actual verification if any path exists will continue to remain in the C++ code.

File: Extension/package.json (hunk 2 above)
Comment (michelleangela) [reply Extension/package.json]: I agree. I'll wait for more feedback from the users and then we can decided on the name and final location.

Comment (bobbrow) [reply Extension/package.json]: I think we're going to go with `%LocalAppData%\Microsoft\vscode-cpptools`

**Pull request merged**